src/components/schedule-visualizer/robot.tsx

```diff
@@ -1,84 +1,87 @@
 import { makeStyles } from '@material-ui/core';
 import * as RomiCore from '@osrf/romi-js-core-interfaces';
-import React from 'react';
+import React, { useMemo, useState, useEffect, useContext } from 'react';
 import ColorManager from './colors';
+import ResourceManager from '../../resource-manager';
+import { ResourcesContext } from '../../app-contexts';
+import RobotDefaultIcon from './robot-default-icon';
+import RobotImageIcon from './robot-image-icon';
 
 const useStyles = makeStyles(() => ({
-  robotMarker: {
-    cursor: 'pointer',
-    pointerEvents: 'auto',
-  },
   robotText: {
     dominantBaseline: 'central',
     textAnchor: 'middle',
     fontSize: '0.18px',
     fontWeight: 'bold',
     fill: 'white',
+    /* 1 pixel black shadow to left, top, right and bottom */
+    textShadow: '-1px 0 black, 0 1px black, 1px 0 black, 0 -1px black',
   },
 }));
 
 export interface RobotProps {
   robot: RomiCore.RobotState;
-  footprint: number;
   colorManager: ColorManager;
+  footprint: number;
+  fleetName: string;
   onClick?(e: React.MouseEvent<SVGGElement>, robot: RomiCore.RobotState): void;
 }
 
 const Robot = React.forwardRef(function(
   props: RobotProps,
   ref: React.Ref<SVGGElement>,
 ): React.ReactElement {
+  const resourcesContext = useContext(ResourcesContext);
   const classes = useStyles();
-  const { robot, footprint, colorManager, onClick } = props;
-  const [robotColor, setRobotColor] = React.useState<string | null>(() =>
-    colorManager.robotColorFromCache(robot.name, robot.model),
-  );
+  const { robot, footprint, colorManager, fleetName, onClick } = props;
+  const [iconError, setIconError] = useState(false);
+  const [renderDefaultIcon, setRenderDefaultIcon] = useState(false);
+
+  // The only image formats SVG software support are JPEG, PNG, and other SVG files.
+  const iconPath = useMemo(() => ResourceManager.getRobotIconPath(resourcesContext, fleetName), [
+    resourcesContext,
+    fleetName,
+  ]);
 
-  React.useLayoutEffect(() => {
-    if (robotColor) {
-      return;
-    }
-    (async () => {
-      setRobotColor(await colorManager.robotColor(robot.name, robot.model));
-    })();
-  }, [robot, robotColor, colorManager]);
+  useEffect(() => {
+    setRenderDefaultIcon(!iconPath || !!iconError);
```

Comment:
can this be a state to avoid double rendering?

Reply:
Makes sense. Done!